xbmc/addons/AddonDatabase.cpp

```diff
@@ -285,6 +290,14 @@ void CAddonDatabase::SyncInstalled(const std::set<std::string>& ids,
           ORIGIN_SYSTEM, id.c_str()));
     }
 
+    for (const auto& id : optional)
+    {
+      m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID='%s'", id.c_str()));
```

Comment:
i wonder if it is correct to always enable optional system addons?

Reply:
Maybe just set the origin if it’s unset and drop the enable statement.

Reply:
Can a bit hard for the user. If he install a addon normally all his dependencies becomes added and he see not easy direct which depends used (there is a dialog, but who looks to them). If them optional and disabled the user mostly not realize this.

Maybe there can be a extra dialog where explain for what they are and ask for use. E.g. on some is inputstream.adaptive optional but for some streams them needed.

Reply:
@AlwinEsch, so by default does that mean any optionally installed addon will be disabled by default?

Reply:
No, them should be enabled otherwise becomes user confused.

Currently with optional can a using addon installed also if the depend not available in mirror.

Reply:
i wonder if there are two types of optional?

for versioncheck, the 'optional' means 'allow user to disable this addon even though it is a system addon':
https://github.com/xbmc/xbmc/pull/11980

then there's the 'optional' attribute in the addon.xml file, which only applies to dependencies
in this case 'optional' means 'allow the addon to be installed even if an optional dependency is missing from the repo'.

Reply:
as this commit is a c+p of what we already do for (non-optional) system addons a few lines above, i think it is no fault to remove the setting to enabled and only update the origin from empty to ORIGIN_SYSTEM. that was the original intention and we don't break any existing behaviour that way. as far as i know `versioncheck` is enabled by default anyway, as long as it's a compatible version. i tend to remove `...SET enabled=1..` later that day.
additionally `AddonDatabse` and `AddonInstaller` will move to `namespace ADDON` for consistency

Reply:
That’s the right decision @howie-f, just set the origin and leave the other behaviour as is.